app/views/submissions/new.html.erb

```diff
@@ -0,0 +1,41 @@
+<div class="show_and_edit">
+  <%= form_for new_post do |post| %>
+    <% if new_post.errors[:base].present? %>
+      <div class="alert alert-alert">
+        <%= show_errors(new_post, :base) %>
+      </div>
+    <% end %>
+    <div class="tabbed">
+      <%= post.label :title %>
+      <%= post.text_field :title %><br/>
+    </div>
+    <% if new_post.errors[:title].present? && !new_post.errors[:base].present? %>
+      <div class="alert alert-alert inline-alert">
+        <%= show_errors(new_post, :title) %>
+      </div>
+    <% end %>
+    <div class="tabbed">
+      <%= post.label :url %>
+      <%= post.url_field :url %><br/>
+    </div>
+    <% if new_post.errors[:url].present? && !new_post.errors[:base].present? %>
+      <div class="alert alert-alert inline-alert">
+        <%= show_errors(new_post, :url) %>
+      </div>
+    <% end %>
+    <div>
+      <p>OR</p>
+      <br/>
+    </div>
+    <div class="tabbed">
+      <%= post.label :text %>
+      <%= post.text_area :text %>
+    </div>
+    <% if new_post.errors[:text].present? && !new_post.errors[:base].present? %>
+      <div class="alert alert-alert inline-alert">
+        <%= show_errors(new_post, :text) %>
+      </div>
+    <% end %>
```

Comment:
There's a common pattern to rendering these errors. Could the helper do more here to encapsulate that pattern?

Reply:
Yes, there's definitely a pattern, but I was thinking of some approach from the Rails Views docs might offer a solution. http://guides.rubyonrails.org/action_view_overview.html#partials

Could something like this - 3.2.5 Rendering Collections - work? Or do you have a different approach?